autoload/OmniSharp/stdio.vim

```diff
@@ -86,31 +80,55 @@ function! s:Request(command, opts) abort
   let buffer = join(lines, sep)
 
   let body = {
-  \ 'Seq': s:nextseq,
-  \ 'Command': a:command,
-  \ 'Type': 'Request',
   \ 'Arguments': {
   \   'Filename': filename,
   \   'Line': lnum,
   \   'Column': cnum,
   \   'Buffer': buffer
   \ }
   \}
-  if has_key(a:opts, 'Parameters')
-    call extend(body.Arguments, a:opts.Parameters, 'force')
+  call s:RawRequest(body, a:command, a:opts, sep)
+endfunction
+
+" body -> request body dict
+" command -> command string. eg: /codecheck
+" opts -> command arugments dict
+" sep -> line separator. Optional, if not given line ending replacemment is not done
+function! s:RawRequest(...) abort
+  let body = get(a:, 1, 0)
+  let command = get(a:, 2, 0)
+  let opts = get(a:, 3, 0)
+  let sep = get(a:, 4, 0)
```

Comment:
Could we make these named parameters? The `...` optional parameters make it a bit harder to follow

```vim
function! s:RawRequest(body, command, opts, ...) abort
  let sep = a:0 ? a:1 : ''
  ...
```

Reply:
Done